source/common/router/router.cc

```diff
@@ -360,6 +390,23 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
     sendNoHealthyUpstreamResponse();
     return Http::FilterHeadersStatus::StopIteration;
   }
+  if (append_cluster_info) {
+    // If we're asked to append cluster info, we now have some additional information to append
+    // on top of the cluster name. The hostname and address will be appended to any local or
+    // upstream responses from this point.
+    modify_headers = [modify_headers, conn_pool](Http::HeaderMap& headers) {
+      modify_headers(headers);
+      headers.addCopy(Http::LowerCaseString("x-envoy-hostname"), conn_pool->host()->hostname());
```

Comment:
I'm not sure what value having a host name is if we don't forward, since a LB might pick a different host next time? Is this in case you are somehow specifying a specific upstream host via some other feature? Should this be an additional configuration to output this data?

Reply:
If we don't forward, the idea is basically to pull an example host from of the LB pool, so you can see where the request might have been forwarded to. This could certainly be enabled by a different flag... I'll think of a good name.